test/command_manager_test.cpp

```diff
@@ -0,0 +1,610 @@
+  }
+  rc_values[2] = 1000;
+
+  float max_roll = rf.params_.get_param_float(PARAM_RC_MAX_ROLL);
+  float max_pitch = rf.params_.get_param_float(PARAM_RC_MAX_PITCH);
+  float max_yawrate = rf.params_.get_param_float(PARAM_RC_MAX_YAWRATE);
+
+  // Let's clear all errors in the state_manager
+  rf.state_manager_.clear_error(rf.state_manager_.state().error_codes);
+
+  //=================================================
+  // RC Arming Test
+  //=================================================
+
+  // Let's send an arming signal
+  rc_values[0] = 1500;
+  rc_values[1] = 1500;
+  rc_values[2] = 1000;
+  rc_values[3] = 2000;
+  board.set_rc(rc_values);
+//  step_firmware(rf, board, 20000);
+  // Step long enough for an arm to happen
+  step_firmware(rf, board, 1100000);
+  // Check the output
+  control_t output = rf.command_manager_.combined_control();
+  EXPECT_EQ(output.x.type, ANGLE);
+  EXPECT_PRETTYCLOSE(output.x.value, 0.0*max_roll);
+  EXPECT_EQ(output.y.type, ANGLE);
+  EXPECT_PRETTYCLOSE(output.y.value, 0.0*max_pitch);
+  EXPECT_EQ(output.z.type, RATE);
+  EXPECT_PRETTYCLOSE(output.z.value, 1.0*max_yawrate);
+  EXPECT_EQ(output.F.type, THROTTLE);
+  EXPECT_PRETTYCLOSE(output.F.value, 0.0);
+
+  // See that we are armed
+  EXPECT_EQ(rf.state_manager_.state().armed, true);
+
+  // Let's send a disarming signal
+  rc_values[0] = 1500;
+  rc_values[1] = 1500;
+  rc_values[2] = 1000;
+  rc_values[3] = 1000;
+  board.set_rc(rc_values);
+  // Step long enough for an arm to happen
+  step_firmware(rf, board, 1200000);
+
+  // See that we are disarmed
+  EXPECT_EQ(rf.state_manager_.state().armed, false);
+  EXPECT_EQ(rf.state_manager_.state().error, false);
+  EXPECT_EQ(rf.state_manager_.state().failsafe, false);
+
+
+  //=================================================
+  // Switch Arming Test
+  //=================================================
+  rf.params_.set_param_int(PARAM_RC_ARM_CHANNEL, 4);
+  rc_values[0] = 1500;
+  rc_values[1] = 1500;
+  rc_values[2] = 1000;
+  rc_values[3] = 1500;
+  rc_values[4] = 1500;
+
+  // Set all stick neutral position
+  board.set_rc(rc_values);
+  step_firmware(rf, board, 20000);
+  // make sure we are still disarmed
+  EXPECT_EQ(rf.state_manager_.state().armed, false);
+  EXPECT_EQ(rf.state_manager_.state().error, false);
+  EXPECT_EQ(rf.state_manager_.state().failsafe, false);
+
+  // flip the arm switch on
+  rc_values[4] = 1900;
+  board.set_rc(rc_values);
+  step_firmware(rf, board, 20000);
+  // we should be armed
+  EXPECT_EQ(rf.state_manager_.state().armed, true);
+  EXPECT_EQ(rf.state_manager_.state().error, false);
+  EXPECT_EQ(rf.state_manager_.state().failsafe, false);
+
+  // flip the arm switch off
+  rc_values[4] = 1100;
+  board.set_rc(rc_values);
+  step_firmware(rf, board, 20000);
+  // we should be disarmed
+  EXPECT_EQ(rf.state_manager_.state().armed, false);
+  EXPECT_EQ(rf.state_manager_.state().error, false);
+  EXPECT_EQ(rf.state_manager_.state().failsafe, false);
```

Comment:
Let's also add the following tests:
  - sticks take the appropriate amount of time to arm
  - sticks don't arm when they shouldn't
  - switch doesn't arm when it shouldn't (maybe just set all of the switch channels high to cover them all and make sure it doesn't arm)